mgmt/rest/rest_v1_test.go

```diff
@@ -0,0 +1,584 @@
+			resp, err := http.Get(
+				fmt.Sprintf("http://localhost:%d/v1/tribe/agreements/%s", r.port, tribeName))
+			So(err, ShouldBeNil)
+			So(resp.StatusCode, ShouldEqual, 200)
+			body, err := ioutil.ReadAll(resp.Body)
+			So(err, ShouldBeNil)
+			So(
+				fmt.Sprintf(fixtures.GET_TRIBE_AGREEMENTS_RESPONSE_NAME),
+				ShouldResemble,
+				string(body))
+		})
+
+		Convey("Get tribe members - v1/tribe/members", func() {
+			resp, err := http.Get(
+				fmt.Sprintf("http://localhost:%d/v1/tribe/members", r.port))
+			So(err, ShouldBeNil)
+			So(resp.StatusCode, ShouldEqual, 200)
+			body, err := ioutil.ReadAll(resp.Body)
+			So(err, ShouldBeNil)
+			So(
+				fmt.Sprintf(fixtures.GET_TRIBE_MEMBERS_RESPONSE),
+				ShouldResemble,
+				string(body))
+		})
+
+		Convey("Delete tribe agreement - v1/tribe/agreements/:name", func() {
+			c := &http.Client{}
+			tribeName := "Agree1"
+			cd := []string{"foo"}
+			body, err := json.Marshal(cd)
+			So(err, ShouldBeNil)
+			req, err := http.NewRequest(
+				"DELETE",
+				fmt.Sprintf("http://localhost:%d/v1/tribe/agreements/%s",
+					r.port,
+					tribeName),
+				bytes.NewReader([]byte{}))
+			So(err, ShouldBeNil)
+			resp, err := c.Do(req)
+			So(err, ShouldBeNil)
+			So(resp.StatusCode, ShouldEqual, http.StatusOK)
+			body, err = ioutil.ReadAll(resp.Body)
+			So(err, ShouldBeNil)
+			So(
+				fmt.Sprintf(fixtures.DELETE_TRIBE_AGREEMENT_RESPONSE_NAME),
+				ShouldResemble,
+				string(body))
+		})
+
+		Convey("Leave tribe agreement - v1/tribe/agreements/:name/leave", func() {
+			c := &http.Client{}
+			tribeName := "Agree1"
+			cd := map[string]string{"Apple": "a", "Ball": "b", "Cat": "c"}
+			body, err := json.Marshal(cd)
+			So(err, ShouldBeNil)
+			req, err := http.NewRequest(
+				"DELETE",
+				fmt.Sprintf("http://localhost:%d/v1/tribe/agreements/%s/leave",
+					r.port,
+					tribeName),
+				bytes.NewReader(body))
+			So(err, ShouldBeNil)
+			resp, err := c.Do(req)
+			So(err, ShouldBeNil)
+			So(resp.StatusCode, ShouldEqual, http.StatusOK)
+			body, err = ioutil.ReadAll(resp.Body)
+			So(err, ShouldBeNil)
+			So(
+				fmt.Sprintf(fixtures.LEAVE_TRIBE_AGREEMENT_REPSONSE_NAME_LEAVE),
+				ShouldResemble,
+				string(body))
+		})
+
+		Convey("Join tribe agreement - v1/tribe/agreements/:name/join", func() {
+			c := &http.Client{}
+			tribeName := "Agree1"
+			cd := cdata.NewNode()
+			cd.AddItem("user", ctypes.ConfigValueStr{Value: "Kelly"})
+			body, err := cd.MarshalJSON()
+			So(err, ShouldBeNil)
+
+			req, err := http.NewRequest(
+				"PUT",
+				fmt.Sprintf("http://localhost:%d/v1/tribe/agreements/%s/join", r.port, tribeName),
+				bytes.NewReader(body))
+			So(err, ShouldBeNil)
+			resp, err := c.Do(req)
+			So(err, ShouldBeNil)
+			So(resp.StatusCode, ShouldEqual, http.StatusOK)
+			body, err = ioutil.ReadAll(resp.Body)
+			So(err, ShouldBeNil)
+			So(
+				fmt.Sprintf(fixtures.JOIN_TRIBE_AGREEMENT_RESPONSE_NAME_JOIN),
+				ShouldResemble,
+				string(body))
+
+		})
+	})
+}
```

Comment:
It is really easy to run all the tests right now; it only takes one command. If we break them up into separate files it will take 3 commands to run all the tests - one command for each section: tribe, task, and metric.